pkg/storage/etcd3/store.go

```diff
@@ -346,19 +356,37 @@ func (s *store) getState(getResp *clientv3.GetResponse, key string, v reflect.Va
 	return state, nil
 }
 
-func (s *store) updateState(st *objState, userUpdate storage.UpdateFunc) (runtime.Object, error) {
-	ret, _, err := userUpdate(st.obj, *st.meta)
+func (s *store) updateState(st *objState, userUpdate storage.UpdateFunc) (runtime.Object, uint64, error) {
+	ret, ttlPtr, err := userUpdate(st.obj, *st.meta)
+
 	version, err := s.versioner.ObjectResourceVersion(ret)
 	if err != nil {
-		return nil, err
+		return nil, 0, err
 	}
 	if version != 0 {
 		// We cannot store object with resourceVersion in etcd. We need to reset it.
 		if err := s.versioner.UpdateObject(ret, 0); err != nil {
 			return nil, fmt.Errorf("UpdateObject failed: %v", err)
 		}
 	}
-	return ret, nil
+	var ttl uint64
+	if ttlPtr != nil {
+		ttl = *ttlPtr
+	}
+	return ret, ttl, nil
+}
+
+// createOpts creates the etcd client options based on given parameters.
+// ttl: if ttl is non-zero, it will attach the key to a lease with ttl of roughly equivalent length
+func (s *store) createOpts(ctx context.Context, ttl int64) ([]clientv3.OpOption, error) {
+	if ttl == 0 {
+		return nil, nil
+	}
+	lcr, err := s.client.Lease.Create(ctx, ttl)
```

Comment:
Regarding @xiang90 concern about this (and @lavalamp comment).
Yes - it's definitely possible that we will have 100`s events per second. Since we are targeting scheduler throughput of 100pods/s (not necessary in 1.3, but still) and we have O(10) events per starting a pod, that means we can even have about 1000 of events per second.

That said, we are not there yet. So we can probably leave as is for now in this PR, add here a TODO and probably also open an issue for it as it will probably hit us. And fixing this in a separate PR is definitely a good idea.